lib/puppet/parser/functions/mysql_password.rb

```diff
@@ -10,6 +10,7 @@ module Puppet::Parser::Functions
     raise(Puppet::ParseError, 'mysql_password(): Wrong number of arguments ' +
       "given (#{args.size} for 1)") if args.size != 1
 
-    '*' + Digest::SHA1.hexdigest(Digest::SHA1.digest(args[0])).upcase
+    # Return the hash or an empty string for an empty argument
+    args[0].length > 0 ? '*' + Digest::SHA1.hexdigest(Digest::SHA1.digest(args[0])).upcase : ''
```

Comment:
```ruby
  return '' if args[0].empty?
  '*' + Digest::SHA1.hexdigest(Digest::SHA1.digest(args[0])).upcase
end
```
strikes me as more readable/rubylike.
(and doesn't need a comment to explain what it does ;)